perceval/backends/mediawiki.py

```diff
@@ -0,0 +1,517 @@
+
+        :returns: a generator of pages
+
+        :raises CacheError: raised when an error occurs accessing the
+            cache
+        """
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_items = self.cache.retrieve()
+
+        pages_dicts = ['allrevisions','allpages','recentchanges']
+
+        # Namespaces -> pages -> revisions
+        next(cache_items) # First namespaces, we don't need them
+        for items in cache_items:
+            data_json = json.loads(items)
+            for pages_dict in pages_dicts:
+                if pages_dict in data_json['query']:
+                    pages_json = data_json['query'][pages_dict]
+                    for page in pages_json:
+                        page_reviews = self.__build_page_reviews(page, json.loads(next(cache_items)))
+                        yield page_reviews
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a wiki page."""
+        return str(item['pageid'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update field from a MediaWiki item.
+
+        The timestamp is extracted from 'timestamp' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        updated = None
+        if item['update']:
+            updated = float(item['update'])
+        return updated
+
+
+    def __get_max_date(self, reviews):
+        """"Get the max date in unixtime format from reviews"""
+        max_ts = 0
+        for review in reviews:
+            ts = str_to_datetime(review['timestamp'])
+            ts = datetime_to_utc(ts)
+            if ts.timestamp() > max_ts:
+                max_ts = ts.timestamp()
+        return max_ts
+
+
+    def __fetch_1_27(self, from_date=None):
+        """Fetch the pages from the backend url for MediaWiki >=1.27
+
+        The method retrieves, from a MediaWiki url, the
+        wiki pages.
+
+        :returns: a generator of pages
+        """
+
+        logger.info("Looking for pages at url '%s'", self.url)
+
+        self._purge_cache_queue()
+        npages = 0  # number of pages processed
+
+        # The pages are organized in content namespaces
+        raw_namespaces = self.client.get_namespaces()
+        namespaces = json.loads(raw_namespaces)["query"]["namespaces"]
+        namespaces_contents = [ns for ns in namespaces if 'content' in namespaces[ns].keys()]
+        self._push_cache_queue(raw_namespaces)
+        self._flush_cache_queue()
+
+        arvcontinue = '' # pagination for getting revisions and their pages
+        while arvcontinue is not None:
+            raw_pages = self.client.get_pages_from_allrevisions(namespaces_contents, from_date, arvcontinue)
+            self._push_cache_queue(raw_pages)
+            self._flush_cache_queue()
```

Comment:
Ok, maybe it is good to flush the cache before all is parsed and yield so if a problem occurs, we have a cache of the already processed items. What do you think? In any case, I will implement now your suggestion. Done!